File: boj/cpp/2609-a.cpp
#include <bits/stdc++.h>
using namespace std;

// 유클리드 호제법
int euclidean_algorithm(int num1, int num2)
{
    while (num2 != 0)
    {
        num1 %= num2;
        swap(num1, num2);
    }
    return num1;
}

int a, b;
int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);

    // 두 자연수를 입력받는다.
    cin >> a >> b;

    int gcd = euclidean_algorithm(a, b);
    cout << gcd << "\n"
         << a * b / gcd;
    return 0;
}

File: boj/java/2609-a.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class Main {

    public void bojSolution() throws Exception {
        // 입력
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int a = Integer.parseInt(st.nextToken());
        int b = Integer.parseInt(st.nextToken());
        br.close();

        // 유클리드 호제법
        class EuclideanAlgorithm {

            private int run(int num1, int num2) {
                if (num2 == 0) {
                    return num1;
                }
                return run(num2, num1 % num2);
            }
        }

        EuclideanAlgorithm euclideanAlgorithm = new EuclideanAlgorithm();
        int gcd = euclideanAlgorithm.run(a, b);

        // 출력
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        bw.write(String.valueOf(gcd));
        bw.newLine();
        bw.write(String.valueOf(a * b / gcd));
        bw.flush();
    }

    public static void main(String[] args) throws Exception {
        new Main().bojSolution();
    }
}